level2/load-balancer.tf

```diff
@@ -0,0 +1,71 @@
+resource "aws_security_group" "lb_sg" {
+  name        = "load balancer sg"
+  description = "Allow HTTP inbound traffic to ELB"
+  vpc_id      = data.terraform_remote_state.level1.outputs.vpc_id
+  ingress {
+    description = "Http to ELB "
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0 //65535
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+
+  }
+
+  tags = {
+    Name = "${var.env_code}lb_sg"
+  }
+}
+
+
+
+resource "aws_lb" "mylb" {
+  name                       = "${var.env_code}lb"
+  internal                   = false
+  load_balancer_type         = "application"
+  security_groups            = [aws_security_group.lb_sg.id]
+  subnets                    = data.terraform_remote_state.level1.outputs.private_subnet_id
```

Comment:
Loadbalancer should be placed on the public subnet. could you please change your subnet to public subnet.